### PR TITLE
New attention metrics to assess attention mask diagonal

### PR DESCRIPTION
**Pull request by pippaduckett1**

### CODE REVIEW

File: src/hepattn/callbacks/attn_mask_logger.py [hunk 1]
```diff
@@ -78,23 +215,27 @@ def _process_attention_masks_from_outputs(self, pl_module, outputs, step, is_val
...
 
+                    # Log diagonal metrics if enabled
+                    if self.log_diagonal_metrics:
+                        self._log_diagonal_metrics(pl_module, attn_mask_im, step, layer_index, f"local_ma_mask_{prefix_suffix}")
+
     def on_validation_batch_end(self, trainer, pl_module, outputs, batch, batch_idx, dataloader_idx=0):
         if not self.log_val:
             return
-        self._process_attention_masks_from_outputs(pl_module, outputs, batch_idx, is_validation=True)
+        self._process_attention_masks_from_outputs(pl_module, outputs, trainer.global_step, is_validation=True)
```
[the remark this answers is not in the document]
Comment (pippaduckett1) [reply src/hepattn/callbacks/attn_mask_logger.py:233]: Yes - the bot before thought that batch_idx for validation would be a problem and create duplicates so I edited to trainer.global_step following it's suggestions (happy to change back if you think this is better though!)
